examples/pump-app/silabs/src/AppTask.cpp

```diff
@@ -0,0 +1,253 @@
+        {
+            sAppTask.DispatchEvent(&event);
+            eventReceived = xQueueReceive(sAppEventQueue, &event, 0);
+        }
+    }
+}
+
+void AppTask::PumpActionEventHandler(AppEvent * aEvent)
+{
+    bool initiated = false;
+    PumpManager::Action_t action;
+    int32_t actor;
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    if (aEvent->Type == AppEvent::kEventType_PumpTimer)
+    {
+        action = static_cast<PumpManager::Action_t>(aEvent->PumpEvent.Action);
+        actor  = aEvent->PumpEvent.Actor;
+    }
+    else if (aEvent->Type == AppEvent::kEventType_Button)
+    {
+        action = (PumpMgr().IsStopped()) ? PumpManager::START_ACTION : PumpManager::STOP_ACTION;
+        actor  = AppEvent::kEventType_Button;
+    }
+    else
+    {
+        err = APP_ERROR_UNHANDLED_EVENT;
+    }
+
+    if (err == CHIP_NO_ERROR)
+    {
+        initiated = PumpMgr().InitiateAction(actor, action);
+
+        if (!initiated)
+        {
+            SILABS_LOG("Action is already in progress or active.");
+        }
+    }
+}
+
+void AppTask::ButtonEventHandler(uint8_t button, uint8_t btnAction)
+{
+    AppEvent button_event           = {};
+    button_event.Type               = AppEvent::kEventType_Button;
+    button_event.ButtonEvent.Action = btnAction;
+
+    if (button == APP_PUMP_SWITCH && btnAction == static_cast<uint8_t>(SilabsPlatform::ButtonAction::ButtonPressed))
+    {
+        button_event.Handler = PumpActionEventHandler;
+        AppTask::GetAppTask().PostEvent(&button_event);
+    }
+    else if (button == APP_FUNCTION_BUTTON)
+    {
+        button_event.Handler = BaseApplication::ButtonHandler;
+        AppTask::GetAppTask().PostEvent(&button_event);
+    }
+}
+
+void AppTask::ActionInitiated(PumpManager::Action_t aAction, int32_t aActor)
+{
+    // Action initiated, update the light led
+    bool pumpOn = aAction == PumpManager::START_ACTION;
+    SILABS_LOG(" Pump %s", (pumpOn) ? "starting" : "stopping")
+
+    sPumpLED.Set(pumpOn);
+
+#ifdef DISPLAY_ENABLED
+    sAppTask.GetLCD().WriteDemoUI(pumpOn);
+#endif
+
+    if (aActor == AppEvent::kEventType_Button)
+    {
+        sAppTask.mSyncClusterToButtonAction = true;
+    }
+}
+
+void AppTask::ActionCompleted(PumpManager::Action_t aAction, int32_t aActor)
+{
+    // action has been completed bon the light
+    if (aAction == PumpManager::STOP_ACTION)
+    {
+        SILABS_LOG("Pump Stopped")
+    }
+    else if (aAction == PumpManager::STOP_ACTION)
+    {
+        SILABS_LOG("Pump OFF")
+    }
+
+    if (sAppTask.mSyncClusterToButtonAction)
+    {
+        chip::DeviceLayer::PlatformMgr().ScheduleWork(UpdateClusterState, reinterpret_cast<intptr_t>(nullptr));
+        sAppTask.mSyncClusterToButtonAction = false;
+    }
+}
+
+void AppTask::UpdateClusterState(intptr_t context)
+{
+    // Set On/Off state
+    EmberStatus status;
+    bool onOffState = !PumpMgr().IsStopped();
+    status          = chip::app::Clusters::OnOff::Attributes::OnOff::Set(1, onOffState);
```

Comment:
you could use the define you created here for the endpoint argument
`#define PCC_CLUSTER_ENDPOINT 1`